process-dataset/data-validation/schema.js

```diff
@@ -25,6 +25,10 @@ const datasetSchema = {
             "description": "Description of the dataset",
             "type": "string",
         },
+        "dateCreated": {
+            "description": "Date the dateset was created (not necessarily the most recent version",
```

Comment:
Typos - should be `Date the dataset was created (not necessarily the most recent version)` (`dateset -> dataset` and missing `)` at the end)

Reply:
Also, does the date need to be in the format you used (Mon DD, YYYY)? If there's a specific format or formats that are accepted I think that should be indicated here?

Reply:
I second the question about the exact date format.  Can it / should it be specified and validated more carefully?

Reply:
Since i'm using the javascript Date object, you can kinda send it in most dates and it will be fine. I added a comment to the description. JSON schema doesn't have a date type. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

Reply:
I won't hold up the PR for this, BUT...

Someone someday is going to put in a bad date, if they are hand-editing, and the code might throw an exception or it might misinterpret it and insert the wrong date.  

The docs you linked actually warn against passing strings to the Date ctor.  So tightening up the requirement will avoid that issue down the line.